src/Query/QueryBuilder.php

```diff
@@ -557,6 +564,56 @@ public function addUnion(string|QueryBuilder $part, UnionType $type = UnionType:
         return $this;
     }
 
+    /**
+     * Add a Common Table Expression to be used for a select query.
+     * Replaces any previously specified parts.
+     *
+     * <code>
+     *     $qb = $conn->createQueryBuilder()
+     *         ->with('cte_a', 'SELECT 1 AS field1');
+     * </code>
+     *
+     * @param string[] $fields The optional fields to select in the CTE.
+     *
+     * @return $this
+     */
+    public function with(string $name, string|QueryBuilder $part, array $fields = []): self
```

Comment:
Please document the meaning of `$name` and the usage of `$fields` in PHPDoc.

Reply:
I think it should be "columns", not "fields". We're not dealing with abstract data structures with fields. We're dealing with relational tables and columns.

Reply:
Please make sure that the term "column" is used consistently in the code being added.